src/linker/Linker.Dataflow/FlowAnnotations.cs

```diff
@@ -28,12 +28,16 @@ public FlowAnnotations (LinkContext context)
 		}
 
 		public bool RequiresDataFlowAnalysis (MethodDefinition method) =>
+			GetAnnotations (method.DeclaringType).TryGetAnnotation (method, out var methodAnnotations)
+				&& (methodAnnotations.ReturnParameterAnnotation != DynamicallyAccessedMemberTypes.None || methodAnnotations.ParameterAnnotations != null);
+
+		public bool HasAnnotations (MethodDefinition method) =>
```

Comment:
I would prefer we don't call this `HasAnnotations` - maybe we could make this more specific:
`RequiresDataFlowAnalysisForVirtualMethods` (based on where this is used).